packages/@apphosting/adapter-nextjs/src/utils.ts

```diff
@@ -42,12 +44,16 @@ export function populateOutputBundleOptions(cwd: string): OutputBundleOptions {
 }
 
 // Run build command
-export function build(cwd: string): void {
+export async function build(cwd: string): Promise<void> {
   // Set standalone mode
   process.env.NEXT_PRIVATE_STANDALONE = "true";
   // Opt-out sending telemetry to Vercel
   process.env.NEXT_TELEMETRY_DISABLED = "1";
-  spawnSync("npm", ["run", "build"], { cwd, shell: true, stdio: "inherit" });
+  if (!(await exists(join(cwd, "node_modules")))) {
```

Comment:
Could we add a more robust test here? Maybe check for a yarn lock? Alt could we expose the package manager via an environment variable in the buildpack?

Reply:
Good call, I changed this to check for yarn.lock.